app/models/manageiq/providers/kubevirt/infra_manager/connection.rb

```diff
@@ -77,7 +90,10 @@ def templates
   # @return [Object] The template object.
   #
   def template(name)
-    @conn.templates.get(name)
+    template = ManageIQ::Providers::Kubevirt::InfraManager::Template.find_by(:name => name)
+    client = namespaced_client(template.location)
+
+    client.templates.get(name)
```

Comment:
Then we don't need to change these getters.  Side note `ManageIQ::Providers::Kubevirt::InfraManager::Template.find_by(:name => name)` isn't guaranteed to be correct since you could have multiple kubevirt providers and templates with the same name.